data/basal.go

```diff
@@ -0,0 +1,166 @@
+		rateTag:         fmt.Sprintf("Must be greater than %f", rateValidationLowerLimit),
+		durationTag:     fmt.Sprintf(validationGreaterThanInt, durationValidationLowerLimit),
+		valueTag:        fmt.Sprintf(validationGreaterThanInt, valueValidationLowerLimit),
+		insulinTag:      fmt.Sprintf("Must be one of %s,%s", levemirInsulin, lantusInsulin),
+	}
+)
+
+//BuildBasal will build a Basal record
+func BuildBasal(obj map[string]interface{}) (*Basal, *Error) {
+
+	base, errs := BuildBase(obj)
+	cast := NewCaster(errs)
+
+	basal := &Basal{
+		DeliveryType: cast.ToString(deliveryTypeField, obj[deliveryTypeField]),
+		ScheduleName: cast.ToString(scheduleNameField, obj[scheduleNameField]),
+		Base:         base,
+	}
+
+	if obj[rateField] != nil {
+		basal.Rate = cast.ToFloat64(rateField, obj[rateField])
+	}
+	if obj[durationField] != nil {
+		basal.Duration = cast.ToInt(durationField, obj[durationField])
+	}
+	if obj[insulinField] != nil {
+		basal.Insulin = cast.ToString(insulinField, obj[insulinField])
+	}
+
+	if validationErrors := validator.Struct(basal); len(validationErrors) > 0 {
+		errs.AppendError(validationErrors.GetError(basalFailureReasons))
+	}
+
+	if errs.IsEmpty() {
+		return basal, nil
+	}
+	return basal, errs
+}
+
+func BasalRateValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if rate, ok := field.Interface().(float64); ok {
+		if rate > rateValidationLowerLimit {
+			return true
+		}
+	}
+	return false
+}
+
+func BasalDurationValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if duration, ok := field.Interface().(int); ok {
+		if duration > durationValidationLowerLimit {
+			return true
+		}
+	}
+	return false
+}
+
+func BasalDeliveryTypeValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if deliveryType, ok := field.Interface().(string); ok {
+		if _, ok = allowedDeliveryTypes[strings.ToLower(deliveryType)]; ok {
+			return true
+		}
+	}
+	return false
+}
+
+func BasalInsulinValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if insulin, ok := field.Interface().(string); ok {
+		if _, ok = allowedInsulins[strings.ToLower(insulin)]; ok {
+			return true
+		}
+	}
+	return false
+}
+
+func BasalValueValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if value, ok := field.Interface().(int); ok {
+		if value > valueValidationLowerLimit {
```

Comment:
Do we need any sort of upper bounds?

What happens if the incoming data has 1.23 for the value.  When we type cast it what happens to the .23?  Is casting the correct solution for numbers?  Perhaps looking at the fieldType?  We don't want to blindly lose data and return success.  It would be far better to fail with an error message "expected integer, got float" (well, better written than that, but you get the point).